src/OpenFeature/Hooks/TraceEnricherHookOptions.cs

```diff
@@ -0,0 +1,91 @@
+
+namespace OpenFeature.Hooks;
+
+/// <summary>
+/// Configuration options for the <see cref="TraceEnricherHookOptions"/>.
+/// </summary>
+public sealed class TraceEnricherHookOptions
+{
+    /// <summary>
+    /// The default options for the <see cref="TraceEnricherHookOptions"/>.
+    /// </summary>
+    public static TraceEnricherHookOptions Default { get; } = new TraceEnricherHookOptions();
+
+    /// <summary>
+    /// Custom dimensions or tags to be associated with current <see cref="System.Diagnostics.Activity"/> in <see cref="TraceEnricherHook"/>.
+    /// </summary>
+    public IReadOnlyCollection<KeyValuePair<string, object?>> CustomDimensions { get; }
+
+    /// <summary>
+    /// Flag metadata callbacks to be associated with current <see cref="System.Diagnostics.Activity"/>.
+    /// </summary>
+    internal IReadOnlyCollection<KeyValuePair<string, Func<ImmutableMetadata, object?>>> FlagMetadataCallbacks { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TraceEnricherHookOptions"/> class with default values.
+    /// </summary>
+    private TraceEnricherHookOptions() : this(null, null)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TraceEnricherHookOptions"/> class.
+    /// </summary>
+    /// <param name="customDimensions">Optional custom dimensions to tag Counter increments with.</param>
+    /// <param name="flagMetadataSelectors">Optional flag metadata callbacks to be associated with current <see cref="System.Diagnostics.Activity"/>.</param>
+    internal TraceEnricherHookOptions(IReadOnlyCollection<KeyValuePair<string, object?>>? customDimensions = null,
+        IReadOnlyCollection<KeyValuePair<string, Func<ImmutableMetadata, object?>>>? flagMetadataSelectors = null)
+    {
+        this.CustomDimensions = customDimensions ?? [];
+        this.FlagMetadataCallbacks = flagMetadataSelectors ?? [];
+    }
+
+    /// <summary>
+    /// Creates a new builder for <see cref="TraceEnricherHookOptions"/>.
+    /// </summary>
+    public static TraceEnricherHookOptionsBuilder CreateBuilder() => new TraceEnricherHookOptionsBuilder();
+
+    /// <summary>
+    /// A builder for constructing <see cref="TraceEnricherHookOptions"/> instances.
+    /// </summary>
+    public sealed class TraceEnricherHookOptionsBuilder
+    {
+        private readonly List<KeyValuePair<string, object?>> _customDimensions = new List<KeyValuePair<string, object?>>();
+        private readonly List<KeyValuePair<string, Func<ImmutableMetadata, object?>>> _flagMetadataExpressions = new List<KeyValuePair<string, Func<ImmutableMetadata, object?>>>();
+
+        /// <summary>
+        /// Adds a custom dimension.
+        /// </summary>
+        /// <param name="key">The key for the custom dimension.</param>
+        /// <param name="value">The value for the custom dimension.</param>
+        public TraceEnricherHookOptionsBuilder WithCustomDimension(string key, object? value)
```

Comment:
What about naming it to `WithTag` since it's going to add tag for the activity

```suggestion
        public TraceEnricherHookOptionsBuilder WithTag(string key, object? value)
```